controllers/front/OrderDetailController.php

```diff
@@ -108,17 +108,31 @@ public function postProcess()
                     }
 
                     if (Validate::isLoadedObject($customer)) {
-                        Mail::Send($this->context->language->id, 'order_customer_comment', Mail::l('Message from a customer'),
-                        array(
-                            '{lastname}' => $customer->lastname,
-                            '{firstname}' => $customer->firstname,
-                            '{email}' => $customer->email,
-                            '{id_order}' => (int)$order->id,
-                            '{order_name}' => $order->getUniqReference(),
-                            '{message}' => Tools::nl2br($msgText),
-                            '{product_name}' => $product_name
-                        ),
-                        $to, $toName, $customer->email, $customer->firstname.' '.$customer->lastname);
+                        Mail::Send(
+                            $this->context->language->id,
+                            'order_customer_comment',
+                            Mail::l('Message from a customer'),
+                            array(
+                                '{lastname}' => $customer->lastname,
+                                '{firstname}' => $customer->firstname,
+                                '{email}' => $customer->email,
+                                '{id_order}' => (int)$order->id,
+                                '{order_name}' => $order->getUniqReference(),
+                                '{message}' => Tools::nl2br($msgText),
+                                '{product_name}' => $product_name
+                            ),
+                            $to,
+                            $toName,
+                            strval(Configuration::get('PS_SHOP_EMAIL')),
```

Comment:
Hi!

Well, besides (as you pointed out) the `Sender:` header which as far as i know is not used by PrestaShop, there is this problem:
In the `From:` field, should only the shop address be included or should it be `Customer Name <shopaddress@example.com>`? In my opinion, it should include the name.

These are a few reasons:

- Shop owners receive mail messages not only in their mailbox, but also in a CRM/management system. Using the customer name can assure proper matching with the customer on file.
- When shops receive messages from customers, it's much easier to deal with a list of conversations with customer names rather than with your shop name for all of them. In this case even when the reply-to field is set, a shop owner cannot tell which emails come from which customers just by looking at the list of conversations.
- As seen in the example above from Github, others major sites do the same: use the their address (notifications@github.com)in the `From:` header, but use the user name. Again in Github's case, the reply-to is set to a automatically generated tracking address.
![screen shot 2017-02-16 at 19 05 59](https://cloud.githubusercontent.com/assets/898057/23034132/031a0eb8-f47b-11e6-8d66-6fc1b3bd0347.png)
- It is a recommended best practice, as shown in the example of the SPF website.

I hope this clarifies the situation ;)